imap/http_jmap.c

```diff
@@ -876,23 +871,28 @@ static int _create_upload_collection(const char *accountid,
 }
 
 HIDDEN int jmap_open_upload_collection(const char *accountid,
-                                       struct mailbox **mailbox)
+                                       struct mailbox **mailboxp)
 {
     /* upload collection */
     mbentry_t *mbentry = NULL;
     int r = lookup_upload_collection(accountid, &mbentry);
-    if (r) {
+
+    if (r == IMAP_MAILBOX_NONEXISTENT) {
         mboxlist_entry_free(&mbentry);
-        return _create_upload_collection(accountid, mailbox);
+        return _create_upload_collection(accountid, mailboxp);
```

Comment:
> > And it means that the uniqueid stays with the INBOX, which is very
> > much needed in uuidland
> 
> I'd like to see a Cassandane test that asserts this behaviour so we don't accidentally break it later
> 
> Some comments inline, I think they might be things I'm not understanding rather than problems

Done: https://github.com/cyrusimap/cassandane/pull/158

Reply:
Ooh yeah good point.  We should just return any other error.  I missed a "goto done".

Reply:
Fix pushed